### DISCUSSION TITLE
Safe to use Pickle with SeleniumBase uc=True

### DISCUSSION THREAD

**Discussion by JonWakefield** [seleniumbase/SeleniumBase#2303] [General]
Hello, I am wondering if it safe to use pickle to store and retrieve cookies all while being undetected. (`driver = Driver(uc=True)`)

### What I am trying to do: 
Currently, when I load up my bot I have to log into the website each time; however if I use pickle and save the cookies I can bypass the login phase.

I am wondering if this is a safe action or if this could potentially lead to me getting caught?

### Here is a brief code snippet of what I am doing:
`
from seleniumbase import Driver
import pickle

// Create a driver
driver = Driver(uc=True)

// log in using cookies:
driver.get(url)

// load cookies:
cookies = pickle.load(open("cookies.pkl", "rb"))

// add cookies
for cookie in cookies:
    driver.add_cookies(cookie)

// Perform additional website actions
// ...
// ...
`

Note: the code works, just wondering if combing pickle with uc=True will lead to me getting detected or if I should still be okay. 

Thanks

**Comment by mdmintz** [comment]
That'll probably still work. You can set a custom `user_data_dir` for UC Mode so that all your existing cookies carry over between your tests. However, if you do set a `user_data_dir`, make sure that the directory is only used by UC Mode Chrome. (Don't mix that directory with your standard Chrome's `user_data_dir`.)

SeleniumBase has special methods for cookies. These let you save and load cookies to/from files:

```python
self.save_cookies(name="cookies.txt")

self.load_cookies(name="cookies.txt")
```

For working with cookies, see https://www.selenium.dev/documentation/webdriver/interactions/cookies/

Here are the other SeleniumBase cookie methods:

```python
self.delete_all_cookies()
# Duplicates: self.clear_all_cookies()

self.delete_saved_cookies(name="cookies.txt")
```

Also, you can preload cookies. (See https://github.com/seleniumbase/SeleniumBase/discussions/2242#discussioncomment-7487896)
